### PR TITLE
Add NodeInformer event handler to volume controller

### PR DESCRIPTION
**Pull request by PhanLe1010**
Make sure to enqueue volume for replica scheduling when there is a change on a node.

longhorn/longhorn#3626

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM

Waiting for @shuo-wu review.

File: controller/volume_controller.go [hunk 1]
```diff
@@ -3459,6 +3466,40 @@ func (vc *VolumeController) enqueueVolumesForBackingImageDataSource(obj interfac
...
+
+		// use the last known state, to enqueue, dependent objects
+		node, ok = deletedState.Obj.(*longhorn.Node)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("DeletedFinalStateUnknown contained invalid object: %#v", deletedState.Obj))
+			return
+		}
+	}
+
+	replicas, err := vc.ds.ListReplicasRO()
```
Comment (shuo-wu) [inline controller/volume_controller.go:3486]: You can have 2 list functions: One is for the unscheduled replicas, another is for the failed replicas **on this node only**.
This is just a tiny suggestion, I am fine with the current implementation as well.

**Review by shuo-wu — COMMENTED** [review]
In general LGTM

**Review by shuo-wu — APPROVED** [review]

File: controller/volume_controller.go (hunk 1 above)
Comment (PhanLe1010) [reply controller/volume_controller.go:3486]: Thanks @shuo-wu ! Good idea as usual! 

I would like to keep the current implementation to verify the theory that the slow enqueueNodeChange function shouldn't affect the scalability

**Comment by innobead** [comment]
@PhanLe1010 please backport this. thanks.

**Comment by PhanLe1010** [comment]
As discussed in Slack, we will change the resync back to 30s for v1.1.4 and v1.2.4 since there are potentially conner cases that we cannot catch before releasing. Those are stable releases so we don't want to have mistakes. Therefore, we don't need to backport this PR to those branches

**Pull request merged**